src/util.c

```diff
@@ -351,7 +351,12 @@ char *git__strsep(char **end, const char *sep)
 
 size_t git__linenlen(const char *buffer, size_t buffer_len)
 {
-	char *nl = memchr(buffer, '\n', buffer_len);
+	char *nl;
+
+	if (!buffer)
```

Comment:
I think this change is sensible. I'd love to have an `assert(buffer || !buffer_len);` here, though, such that we can catch programming errors in case anybody calls `git__linenlen(NULL, n)` with `n > 0`.

Reply:
I added an assertion.